### ISSUE TITLE
Add command /random

### ISSUE THREAD

**Issue by lolipodass** [ddnet/ddnet#8088]
add the ability to change the map through the command, not through the UI. Since at the moment this is one of the few (if not the only) action that cannot be done through the CLI. 
Also, this command can be added advanced functionality, like query parameters.
For example /random medtime<200 starts>2 stars<4 finishes>200 and so on. This command will allow much easier to find maps on such filters, which is now quite difficult to do, such as brutals, which are already more than 400, and find something on a similar condition will be quite long.

**Comment by heinrich5991** [comment]
You can already call the vote through f1 using `call_vote`.

**Comment by lolipodass** [comment]
> You can already call the vote through f1 using `call_vote`.

it's not exactly the same thing. First of all, it uses a different console than all other server related commands. Also because of this command will not be in /cmdlist. Also, about this possibility in this case knows a very small number of people. And most often people go to the menu>call vote>change the map>did not like it and change again, and so may be 5-10 that it becomes tedious.

**Comment by lolipodass** [comment]
> How do we educate people that they can start votes via the console?

You don't even have to teach people much, since people in this game learn almost everything non-verbally, but here are a few basic ways that come to mind.
1 through the description in the vote, such as the command /map, which I just learned about in this way
2 through the words of other people, like I learned about /tpxy, /deep and told others about /tc /tp and others.
3 through cmdlist, some people look at it sometimes (like me), and from there I learned about /times for example
4 description in changelog with an example (screenshot), this method will work very well if the second part of the command, query params, is implemented.

**Comment by heinrich5991** [comment]
No no, I mean teaching them about the existing method. I don't feel that implementing a complex filtering should be done, seems out of scope for the server. It'd be better if someone could create a website where you can filter the maps like you want.

**Comment by lolipodass** [comment]
> No no, I mean teaching them about the existing method. I don't feel that implementing a complex filtering should be done, seems out of scope for the server. It'd be better if someone could create a website where you can filter the maps like you want.

to be honest, then at least add the ability to simply change the map, because now to do it through f1 it takes even more time than through the ui. 
I think that the basic implementation of /random 5(star dif) will be at least enough.
here are my attempts to do a callvote without the knowledge. 

![image](https://github.com/ddnet/ddnet/assets/56706038/6322aa81-5068-44fb-9cef-ea28f16ecf8f)

**Comment by heinrich5991** [comment]
`callvote option "Random Brutal Map (Reason=Stars)" 5`

**Comment by kaitlynia** [comment]
@lolipodass
> Yes, I after the third attempt to understand how to do it, but to write all this is very tedious + it will work only for Brutal, if you need another category, you need to rewrite again, and for example write for ddmax.next pro and so on, and if unfinished, so it is generally formed a very large line, dont forget order of words and so on, which is really not user friendly.

I agree with this. It could be made simpler if the text was always the same, "Random Map In Category (Reason=Stars)" or something like this. Obviously this is not as readable in the UI, but I remember as a new player reading "Random Novice Map Unfinished by Vote Caller" and having zero clue what that meant until I clicked it, so maybe it doesn't matter much

**Comment by AssassinTee** [comment]
I don't think commands should be the goto-way a user should switch maps (`random` would be a bad command name). Since we have #10819 now, I don't find this feature useful. In the end you'll **never** know if the map that the query results in is one you like. Querying for the number of finishes is also a bad idea, there are fun maps for example which are just unplayed.

Closed in favor of #10819

**Comment by lolipodass** [comment]
> I don't think commands should be the goto-way a user should switch maps (`random` would be a bad command name). Since we have [#10819](https://github.com/ddnet/ddnet/pull/10819) now, I don't find this feature useful. In the end you'll **never** know if the map that the query results in is one you like. Querying for the number of finishes is also a bad idea, there are fun maps for example which are just unplayed.
> 
> Closed in favor of [#10819](https://github.com/ddnet/ddnet/pull/10819)

no, as I said in the first message, the main idea is just to implement the random function in a category through a chat command, and filters are an ideal case.
The main reason why the chat function is needed is because it takes too long to find a random matching map, because now
esc->call vote->select random->look at map, but now starting interesting part
esc->call vote ->wait like 3-5 seconds (because if you click too fast, you won't the list has loaded completely, or because of something else it happens, your click will not work and you need to start from the beginning )->select random->look at map

I do not know if / random will solve the problem of waiting until something loads, but at least if not, then the command can at least be spammed until it works, and not wait for some incomprehensible time and then just click on the menu item.

and I just suggest name /random as first that came to mind, it's can be any name

**Comment by AssassinTee** [comment]
>  too long to find a random matching map

Then a random command also won't help you, because commands are also spam protected. I refuse the added complexity of adding a command in order to remove like 3 seconds from the voting process.

If searching for a map over voting is too slow to you, then please use the website (for example) https://ddnet.org/ranks/brutal/

I must admit, that the website is also not optimal in this regard
